tests/test_mailchimp_bookmarks.py

```diff
@@ -0,0 +1,240 @@
+
+                        self.assertLessEqual(
+                            replication_key_value,
+                            first_bookmark_value_ts,
+                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced.",
+                        )
+
+                    for record in second_sync_messages:
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                        replication_key_value = self.parse_date(
+                            record.get(replication_key), self.RECORD_DATETIME_FORMAT
+                        )
+                        self.assertGreaterEqual(
+                            replication_key_value,
+                            simulated_bookmark_value,
+                            msg="Second sync records do not respect the previous bookmark.",
+                        )
+
+                        # Verify the second sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(
+                            replication_key_value,
+                            second_bookmark_value_ts,
+                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced.",
+                        )
+
+                    # Verify that you get less data the 2nd time around
+                    self.assertLess(
+                        second_sync_count,
+                        first_sync_count,
+                        msg="second sync didn't have less records, bookmark usage not verified",
+                    )
+
+                elif expected_replication_method == self.FULL_TABLE:
+
+                    # Verify the syncs do not set a bookmark for full table streams
+                    self.assertIsNone(first_bookmark_value)
+                    self.assertIsNone(second_bookmark_value)
+
+                    # Verify the number of records in the second sync is the same as the first
+                    self.assertEqual(second_sync_count, first_sync_count)
+
+                    # Verify that for the full table stream, all data replicated in sync 1 is replicated again in sync 2
+                    for record in second_sync_messages:
+                        self.assertIn(record, first_sync_messages)
+                else:
+
+                    raise NotImplementedError(
+                        "INVALID EXPECTATIONS\t\tSTREAM: {} REPLICATION_METHOD: {}".format(
+                            stream, expected_replication_method
+                        )
+                    )
+
+                # Verify at least 1 record was replicated in the second sync
+                self.assertGreater(
+                    second_sync_count,
+                    0,
+                    msg="We are not fully testing bookmarking for {}".format(stream),
+                )
```

Comment:
It seems unreachable assertion for `INCREMENTAL` streams. Should this be first assertion in the validations?

Reply:
This assertion was outside both the conditions, so it was being tested for both the streams, however i  have moved it above to be only tested for incremental streams.

should we test this condition for both incremental as well as full table streams?